### ISSUE TITLE
task_definition version always known after apply

### ISSUE THREAD

**Issue by TarekAS** [terraform-aws-modules/terraform-aws-atlantis#72]
Hello,

While using this module, I get the following message every time I do ```terraform apply```.

```
~ resource "aws_ecs_service" "atlantis" {
        cluster                            = "arn:aws:ecs:eu-west-1:[REDACTED]:cluster/atlantis"
        deployment_maximum_percent         = 200
        deployment_minimum_healthy_percent = 50
        desired_count                      = 1
        enable_ecs_managed_tags            = false
        health_check_grace_period_seconds  = 0
        iam_role                           = "aws-service-role"
        id                                 = "arn:aws:ecs:eu-west-1:[REDACTED]:service/atlantis"
        launch_type                        = "FARGATE"
        name                               = "atlantis"
        platform_version                   = "LATEST"
        propagate_tags                     = "NONE"
        scheduling_strategy                = "REPLICA"
        tags                               = {}
      ~ task_definition                    = "atlantis:2" -> (known after apply)
```

Is there any way to avoid this?

The following is my module usage:

```
module "atlantis" {
  source  = "terraform-aws-modules/atlantis/aws"
  version = "~> 2.0"

  name               = "atlantis"
  vpc_id             = "${aws_vpc.myvpc.id}"
  private_subnet_ids = ["${aws_subnet.private-1a.id}", "${aws_subnet.private-1b.id}", "${aws_subnet.private-1c.id}"]
  public_subnet_ids  = ["${aws_subnet.public-1a.id}", "${aws_subnet.public-1b.id}", "${aws_subnet.public-1c.id}"]

  certificate_arn       = "redacted"
  create_route53_record = false

  atlantis_github_user       = "${local.atlantis_github_username}"
  atlantis_github_user_token = "${local.atlantis_github_access_token}"
  atlantis_repo_whitelist    = ["redacted"]

  ssm_kms_key_arn = "redacted"
}
```

**Comment by jondunning** [comment]
Having the same issue

**Comment by jseiser** [comment]
Same issue present in my deployments as well.

**Comment by wreulicke** [comment]
I have same issue.

I seem that this issue is caused by this PR.
https://github.com/terraform-aws-modules/terraform-aws-atlantis/pull/62

**Comment by wreulicke** [comment]
@antonbabenko 
What do you think?

**Comment by tomwj** [comment]
https://github.com/tomwj/terraform-aws-atlantis/commit/22a4b644bcc14447ca6d209013e8055c88561d21

 Fix issue #72 task_definition known after apply

Using a data provider prevents terraform converging. Basically until the apply runs it doesn't have the data to know which revision is great. Forcing it to be updated on every apply. It doesn't have the data available since it's pulling from AWS when it has it locally already.

It does mean if you update the task_definition in the AWS console this will overwrite it. Which I believe is correct behaviour. e.g. I bumped the task_definition manually to 16 and it will roll it back.
```
    ~ task_definition                    = "atlantis:16" -> "atlantis:15"
```
 https://www.terraform.io/docs/configuration/data-sources.html#data-resource-dependencies

**Comment by thyming** [comment]
does anyone have a workaround for this issue?

**Comment by rotarur** [comment]
I have the same issue. i think `depends_on` can be safely removed as it is referenced in another resource to be used.
I did it in my tests and the plan doesn't show any changes:

```
data "aws_ecs_task_definition" "atlantis" {
   task_definition = var.name
}
```

**Comment by rpdelaney** [comment]
Looks like this was fixed in #163
